### PR TITLE
[C#] Extract concepts from v2 change exercise

### LINKED ISSUE #259: [C#] Extract Concepts from v2 exercise: change
Extract the list of concepts required to solve the v2 C# exercise: change. For each concept, describe why it is required to solve the exercise. Keep in mind that an exercise can have multiple approaches. The goal of this task is to help determine the total list of concepts that make up C#.

The list of concepts should be stored in the following document: `languages/csharp/reference/exercise-concepts/change.md`. [Use this link to create a PR to add this document][pr-link].

[pr-link]: ../new/master?filename=languages/csharp/reference/exercise-concepts/change.md

### PR DESCRIPTION
**Pull request by ErikSchierboom**
Closes #259

### CODE REVIEW

File: languages/csharp/reference/exercise-concepts/change.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Concepts of change
+
+[Example implementation](https://github.com/exercism/csharp/blob/master/exercises/change/Example.cs)
+
+## General
+
+- functions: used as the main entry point for the exercise
+- function arguments: input strand is passed as an arguments
```
Comment (robkeim) [inline languages/csharp/reference/exercise-concepts/change.md]:
```suggestion
- function arguments: input coins and target are passed as arguments
```

File: languages/csharp/reference/exercise-concepts/change.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+- conditionals using if: conditionally execute logic using an `if` statement and the ternary operator
+- exceptions: throw an exception in the event of invalid input
+- scoping: use `{` and `}` to denote scoping
+- classes: the tested method is defined in a class
+- objects: creating an object to keep track of the coins
+- members: methods linked to a class instance
+- visibility: making tested method and tested class `public`
+- imports: import types through `using` statements
+- namespaces: knowing where to find the `List<T>` and `Dictionary<TKey, TValue>` classes
+- type inference: using `var` to define the score
```
Comment (robkeim) [inline languages/csharp/reference/exercise-concepts/change.md]:
```suggestion
- type inference: using `var` to define the complex types (dictionaries)
```

File: languages/csharp/reference/exercise-concepts/change.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+- assignment: assigning values, such as the score
```
Comment (robkeim) [inline languages/csharp/reference/exercise-concepts/change.md]:
```suggestion
- assignment: assigning values, such as the minimalCoinsMap
```

File: languages/csharp/reference/exercise-concepts/change.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+- mutation: mutating the list after it has been created
```
Comment (robkeim) [inline languages/csharp/reference/exercise-concepts/change.md]:
```suggestion
- mutation: mutating the dictionary after it has been created
```

File: languages/csharp/reference/exercise-concepts/change.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+- mapping, selecting, ordering enumerables: using multiple LINQ methods to manipulate enumerables (`Min`, `Where`, `OrderBy`, `Select`, `Aggregate`, `FirstOrDefault`, `ToList`, `ToArray`)
+- ranges: a range is created for iterating over all values up to the target
+- object initializers: initializing dictionary through object initializer
+- integers: an `int` is used for a coin and the target
+- array: a `int []` is used for the available coinage
+- lists: a `List<int>` is used as to store the rolls in
```
Comment (robkeim) [inline languages/csharp/reference/exercise-concepts/change.md]:
```suggestion
- lists: a `List<int>` is used as to store the coins for a given value in
```

**Comment by ErikSchierboom** [comment]
I've updated @robkeim!

**Comment by ErikSchierboom** [comment]
Thanks @robkeim!

**Pull request merged**